app/models/provider_label_tag_mapping.rb

```diff
@@ -0,0 +1,96 @@
+class ProviderLabelTagMapping < ApplicationRecord
+  belongs_to :tag
+  belongs_to :mappable_entity
+
+  # Return a flat list of mappable entity names. The nil is required by the
+  # user interface drop down menu for "all".
+  #
+  def self.mappable_entities(provider = 'kubernetes')
+    MappableEntity.where(:provider => provider).pluck(:name).sort.unshift(nil)
+  end
+
+  # Pass the data this returns to map_* methods.
+  def self.cache
+    # {[name, type, value] => [tag_id, ...]}
+    in_my_region.find_each
+                .group_by { |m| [m.label_name, m.labeled_resource_type, m.label_value].freeze }
+                .transform_values { |mappings| mappings.collect(&:tag_id) }
```

Comment:
I'm not positive yet, but I think that these hash entries will need to expand to include the `provider`.  Since `label_name`, `labeled_resource_type` and `label_value` won't be enough to distinguish tag mappings from one type of provider from another.